src/flinkStatementResultsManager.test.ts

```diff
@@ -219,4 +259,83 @@ describe("FlinkStatementResultsManager", () => {
       );
     });
   });
+
+  it("should handle GetStatementMeta message", async () => {
+    await createResultsManagerWithResults();
+
+    const meta = manager.handleMessage("GetStatementMeta", {});
+    assert.deepStrictEqual(meta, {
+      name: mockStatement.name,
+      status: mockStatement.status?.phase,
+      startTime: mockStatement.metadata?.created_at,
+      detail: mockStatement.status?.detail ?? null,
+      failed: mockStatement.failed,
+      stoppable: mockStatement.stoppable,
+      isResultsViewable: mockStatement.isResultsViewable,
+    });
+  });
+
+  it("should handle StopStatement message with retries", async () => {
+    await createResultsManagerWithResults();
+
+    // Mock the updateSqlv1Statement to fail with 409 twice then succeed
+    flinkSqlStatementsApi.updateSqlv1Statement
+      .onFirstCall()
+      .rejects(createResponseError(409, "Conflict", "test"));
+    flinkSqlStatementsApi.updateSqlv1Statement
+      .onSecondCall()
+      .rejects(createResponseError(409, "Conflict", "test"));
+    flinkSqlStatementsApi.updateSqlv1Statement.onThirdCall().resolves();
+
+    await manager.handleMessage("StopStatement", {});
+
+    await eventually(() => {
+      assert.equal(flinkSqlStatementsApi.updateSqlv1Statement.callCount, 3);
+    });
+  });
+
+  it("should handle StopStatement message with max retries exceeded", async () => {
+    await createResultsManagerWithResults();
+
+    // Mock the updateSqlv1Statement to always fail with 409
+    const responseError = createResponseError(409, "Conflict", "test");
+    flinkSqlStatementsApi.updateSqlv1Statement.rejects(responseError);
+
+    // Call stop statement and expect it to throw after max retries
+    await manager.handleMessage("StopStatement", {});
+    assert.equal(flinkSqlStatementsApi.updateSqlv1Statement.callCount, 5);
+  });
+
+  it("should stop polling when statement is not results viewable", async () => {
+    await createResultsManagerWithResults();
+
+    assert.ok(manager["_pollingInterval"] as NodeJS.Timeout);
+
+    const nonViewableStatement = new FlinkStatement({
+      ...mockStatement,
+      status: {
+        ...mockStatement.status,
+        phase: Phase.FAILED,
+        detail: "Statement failed",
+      },
+    });
+    refreshFlinkStatementStub.returns(Promise.resolve(nonViewableStatement));
+
+    // Verify polling was stopped
+    await eventually(() => {
+      assert.equal(manager["_pollingInterval"], undefined);
+    });
+  });
+
+  it("should handle non-409 errors in StopStatement immediately", async () => {
+    await createResultsManagerWithResults();
+
+    flinkSqlStatementsApi.updateSqlv1Statement.rejects(
+      createResponseError(500, "Server Error", "test"),
+    );
+
+    await manager.handleMessage("StopStatement", {});
+
+    assert.equal(flinkSqlStatementsApi.updateSqlv1Statement.callCount, 1);
```

Comment:
Does this actually check that it handled the error?

Reply:
No, I couldn't find an easy way to test that. Perhaps the test not failing unexpectedly itself is a good proof that it handled it.